src/GraphQL/Subscription/ObservableExtensions.cs

```diff
@@ -0,0 +1,253 @@
+namespace GraphQL.Subscription;
+
+internal static class ObservableExtensions
+{
+    /// <summary>
+    /// <para>
+    /// Applies an asynchronous transformation on data events from an <see cref="IObservable{T}"/>.
+    /// Maintains the order of the events produced by the <see cref="IObservable{T}"/>
+    /// whether they are data, error or completion notifications.
+    /// </para>
+    /// <para>
+    /// Ensures that after an <see cref="IObserver{T}"/> stream has been disposed,
```

Comment:
IObserver stream or IObservable stream?

Reply:
I'm not sure exactly what terminology here.  I want to indicate the resulting `IDisposable` returned from `IObservable.Subscribe`, not the `IObservable` itself, but nor is it the `IObserver` either.

Reply:
Resulting `IDisposable` from `Subscribe` method is called _subscription_.

Reply:
```suggestion
    /// Ensures that after a subscription has been disposed,
```